### ISSUE TITLE
Indexing using wpcli prematurly terminated

### ISSUE THREAD

**Issue by ivankristianto** [10up/ElasticPress#714]
I have 950 posts.

I run:
```
vagrant@vvv:/srv/www/10up/htdocs$ wp elasticpress index --setup
Adding mapping...
Deleting index...
Success: Index deleted
Success: Mapping sent
Indexing posts...
Request string length: 1.65 MB
Processed 350/950 entries. . .
Request string length: 1.48 MB
Processed 700/950 entries. . .
vagrant@vvv:/srv/www/10up/htdocs$
```

The `wp elasticpress stats` return:
====== Stats for: 10updev-1 ======
Documents:  700
Index Size: 3.60 MB
====== End Stats ======

Environment:
WP 4.7.2
EP 2.2
ES 5.0.0

**Comment by ivankristianto** [comment]
Stack trace:

```
PHP Fatal error:  Uncaught Error: Call to a member function get_permalink() on boolean in /srv/www/10up/htdocs/wp-content/plugins/woocommerce/includes/class-wc-post-data.php:55
Stack trace:
#0 /srv/www/10up/htdocs/wp-includes/class-wp-hook.php(300): WC_Post_Data::variation_post_link('http://10up.dev...', Object(WP_Post))
#1 /srv/www/10up/htdocs/wp-includes/plugin.php(203): WP_Hook->apply_filters('http://10up.dev...', Array)
#2 /srv/www/10up/htdocs/wp-includes/link-template.php(298): apply_filters('post_type_link', 'http://10up.dev...', Object(WP_Post), false, false)
#3 /srv/www/10up/htdocs/wp-includes/link-template.php(148): get_post_permalink(Object(WP_Post), false, false)
#4 /srv/www/10up/htdocs/wp-content/plugins/ElasticPress/classes/class-ep-api.php(585): get_permalink(Object(WP_Post))
#5 /srv/www/10up/htdocs/wp-content/plugins/ElasticPress/classes/class-ep-api.php(2408): EP_API->prepare_post(42)
#6 /srv/www/10up/htdocs/wp-content/plugins/ElasticPress/bin/wp-cli.php(576): ep_prepare_post(42)
#7 /srv/www/10up/htdocs/wp-cont in /srv/www/10up/htdocs/wp-content/plugins/woocommerce/includes/class-wc-post-data.php on line 55
```

related: #694

**Comment by ivankristianto** [comment]
Just to add another information:
this only happens when it has pagination or posts > 350,
with a woocommerce products that has variations.

**Comment by ivankristianto** [comment]
Doing more debug in depth.
Get to the point what cause it.
This line will reset the $wp_actions:
https://github.com/10up/ElasticPress/blob/develop/bin/wp-cli.php#L840

Thus will cause an error from the woocomerce data, when we get the permalink.

Need advice to fix this.

**Comment by tlovett1** [comment]
Mind writing a PR to fix?

**Comment by tlovett1** [comment]
Also, what test data are you using?

**Comment by ivankristianto** [comment]
WordPress theme-unit-testing.xml, Woocommerce dummy products, and using fakerpress to add 600 posts.

**Comment by tlovett1** [comment]
Somehow I'm not able to reproduce. I am using WC dummy products from here: https://docs.woocommerce.com/document/importing-woocommerce-dummy-data/

**Comment by michaelrimbach** [comment]
I'm having the same bug.

**Comment by ivankristianto** [comment]
related and fixed for: #694
